File: 프로그래머스/unrated/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
    var answer = [];
    var numer = numer1 * denom2 + numer2 * denom1;
    var denom = denom1 * denom2;
    
    let small = Math.min(numer, denom);
    let big = Math.max(numer,denom);    
    while (small != 0)
    {
        const tmp = small;
        small = big % small;
        big = tmp;
    }
    answer.push(numer / big);
    answer.push(denom / big);
    return answer;        
}

File: 프로그래머스/unrated/120808. 분수의 덧셈/분수의 덧셈.c
    
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int numer1, int denom1, int numer2, int denom2) {
    
    int numer = numer1 * denom2 + numer2 * denom1;
    int denom = denom1 * denom2;
    
    int small = (denom < numer ? denom : numer);
    int big = (small == denom ? numer : denom);

    while (small != 0)
    {
        int tmp = small;
        small = big % small;
        big = tmp;
    }
    int* answer = (int*)malloc(2 * sizeof(int));
    answer[0] = (numer / big);
    answer[1] = (denom / big);
    return answer;
}